### PR TITLE
Add homework files

### PR DESCRIPTION
**Pull request by gulyener**
I only used a txt file to store the to-dos for now

### CODE REVIEW

File: week2/homework/src/help.txt [hunk 1]
```diff
@@ -0,0 +1,25 @@
+  
+  Hello there!
```
Comment (gajduk) [inline week2/homework/src/help.txt]: 😄

File: week2/homework/src/index.js [hunk 2]
```diff
@@ -1,3 +1,34 @@
 'use strict';
 
 // TODO: Write the homework code in this file
+{
+  const fs = require('fs');
+  // const showHelp = require('./help.txt');
+  const add = require('./add');
```
Comment (gajduk) [inline week2/homework/src/index.js]: separate files 👍

File: week2/homework/src/remove.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+const fs = require('fs');
+
+function remove(todoTask) {
+  if (todoTask !== undefined) {
+    fs.readFile('./data.txt', 'utf-8', (err, data) => {
+      if (err) throw err;
```
Comment (gajduk) [inline week2/homework/src/remove.js]: Imagine trying to like post of a friend on Instagram and seeing the following error:
```
Error: Cannot write permissions of '\\Home-pc\e\MA MUSIQUE\DIVERS\data.txt.
Windows Error Code 1314: A required privilege is not held by the client. 
```

You were just trying to like a post what does this have to do with files and privileges. That is how the user of the todo app will feel. Here is an example of an uncaught error showing up on a dutch government website
image
and here is one where the stacktrace is printed on an invoice
image

Conclusion: always wrap your errors.

File: week2/homework/src/remove.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+      const newValue = data.replace(todoTask, '');
+
+      fs.writeFile('./data.txt', newValue, 'utf-8', err => {
+        if (err) throw err;
+        console.log(`${todoTask} removed from the list`);
```
Comment (gajduk) [inline week2/homework/src/remove.js]: what if the task is not in the list?

File: week2/homework/src/add.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+const fs = require('fs');
+
+function add(todoTask) {
+  if (todoTask === undefined) {
+    console.log('Please fill in a legitimate task to add');
+  } else {
+    fs.appendFile('./data.txt', `${todoTask}\n`, error => {
+      if (error) throw error;
+    });
+    console.log(`${todoTask} added to the list`);
```
Comment (gajduk) [inline week2/homework/src/add.js]: This should probably be inside the callback

File: week2/homework/src/remove.js (hunk 3 above)
Comment (gulyener) [reply week2/homework/src/remove.js]: Thanks for the detailed explanation and examples @gajduk.  I made a small change on how I handle the errors. I don't know if this is enough or if I need more exhaustive error handling.

File: week2/homework/src/remove.js (hunk 4 above)
Comment (gulyener) [reply week2/homework/src/remove.js]: Thanks I added another function to check if the task exists

File: week2/homework/src/add.js (hunk 5 above)
Comment (gulyener) [reply week2/homework/src/add.js]: Yes you are right I removed it

**Comment by gulyener** [comment]
Thank you for the review and your time and sorry about the late reply. Also thanks for teaching us.

File: week2/homework/extra-hw/index.js [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  res.status = 201;
+  res.end('New counter is added to the list');
+}
+
+async function updateCounter(req, res) {
+  const updatedCounter = validateCounter(req);
+  const counters = await readFromFile();
+  const originalCounter = counters.find(counter => counter.id === req.params.id);
+  originalCounter.name = updatedCounter.name;
+  updatedCounter.name = req.body.counter.name;
```
Comment (gajduk) [inline week2/homework/extra-hw/index.js]: 😕

File: week2/homework/src/remove.js [hunk 7]
```diff
@@ -1,15 +1,26 @@
...
     fs.readFile('./data.txt', 'utf-8', (err, data) => {
-      if (err) throw err;
+      if (err) throw Error('File cannot be read');
 
       const newValue = data.replace(todoTask, '');
 
       fs.writeFile('./data.txt', newValue, 'utf-8', err => {
-        if (err) throw err;
-        console.log(`${todoTask} removed from the list`);
+        if (err) throw Error('File cannot be written');
```
Comment (gajduk) [inline week2/homework/src/remove.js]: 👍

**Pull request closed without merging**